### PR TITLE
refactor: MRMTransitionGroupPicker

### PR DESCRIPTION
**Pull request by pcolaianni**
This should better reflect what you asked at our last meeting.

I invite you to test the code, and read the few TODOs that I've added. Looking at the tests that fail, you'll notice that the culprit is a different value for the RT. If the changes look reasonable to you, I will update all the test cases and propose the PR to _upstream_.
EDIT: found the reason why the tests were failing (transition RT / group RT problem).

I added the information about `start_position_at_50` and also the `end_position_at_50` inside the features.
- Should I rename those strings to `position` or leave them as `time`? EDIT: rename to position

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h [hunk 1]
```diff
@@ -304,11 +305,10 @@ namespace OpenMS
...
 
-        ConvexHull2D::PointArrayType hull_points;
-        double intensity_sum(0.0), intensity_integral(0), rt_sum(0.0);
-        double peak_apex_int = -1;
-        calculatePeakApexInt_(used_chromatogram, best_left, best_right, hull_points, intensity_sum, intensity_integral, rt_sum, peak_apex_int, peak_apex);
-
+        PeakIntegrator::PeakArea pa = pi_.integratePeak(used_chromatogram, best_left, best_right);
+        double intensity_sum = pa.area; // TODO: rename intensity_sum, it doesn't really describe the type of area calculated
+        double peak_apex_int = pa.height;
+        peak_apex = pa.apex_pos; // TODO: should I update this value?
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h]: `apex_pos` is much better I think.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h [hunk 2]
```diff
@@ -304,11 +305,10 @@ namespace OpenMS
         f.setQuality(0, quality);
         f.setOverallQuality(quality);
 
-        ConvexHull2D::PointArrayType hull_points;
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h]: We want to be able to capture the `hull_points`.  If this is missing in `PeakIntegrator::integratePeak` we will need to add it in as an option.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h [hunk 3]
```diff
@@ -317,13 +317,11 @@ namespace OpenMS
...
               "Tried to calculate background estimation without any smoothed chromatograms");
           }
-          else if (background_subtraction_ == "original")
+          else
           {
-            calculateBgEstimationAverage_(used_chromatogram, best_left, best_right, background, avg_noise_level);
-          }
-          else if (background_subtraction_ == "exact")
-          {
-            calculateBgEstimationExact_(used_chromatogram, best_left, best_right, peak_apex_int, background, avg_noise_level);
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h]: Double check that you updated for all places that this function is called (and the other functions) that are being replaced by PeakIntegrator.

**Review by dmccloskey — COMMENTED** [review]
The really reduced the amount of code in MRMTransitionGroupPicker.  This looks lot better.

**Comment by pcolaianni** [comment]
@dmccloskey We need to take a look at the `.featureXML` file generated by the test _UTILS_MRMTransitionGroupPicker_test_1_out_
This is the only test not passing.

**Pull request closed without merging**